components/PromptChangePassword.vue

```diff
@@ -0,0 +1,345 @@
+import { CHARSET, randomStr } from '@/utils/string';
+import CopyToClipboard from '@/components/CopyToClipboard.vue';
+
+export default {
+  components: {
+    ButtonGroup, Checkbox, Card, AsyncButton, LabeledInput, CopyToClipboard, Banner
+  },
+  data(ctx) {
+    return {
+      errorMessages:              [],
+      canShowMissmatchedPassword: false,
+      isUserGenerated:            true,
+      genPasswordOptions:         [{
+        labelKey: `prefs.account.changePassword.userPassword`,
+        value:    true
+      }, {
+        labelKey: `prefs.account.changePassword.generatedPassword`,
+        value:    false
+      }],
+      form: {
+        deleteKeys: false,
+        currentP:   null,
+        newP:       null,
+        genP:       randomStr(16, CHARSET.ALPHA_NUM),
+        confirmP:   null
+      }
+    };
+  },
+  computed:   {
+    ...mapGetters({ t: 'i18n/t' }),
+
+    passwordType: {
+      get() {
+        return this.isUserGenerated;
+      },
+
+      set(isUserGenerated) {
+        this.isUserGenerated = isUserGenerated;
+        if (isUserGenerated) {
+          this.validateNewPassword();
+        } else {
+          this.errorMessages = [];
+        }
+      }
+    },
+
+    passwordNew: {
+      get() {
+        return this.form.newP;
+      },
+
+      set(p) {
+        this.form.newP = p;
+        this.validateNewPassword();
+      }
+    },
+
+    passwordConfirm: {
+      get() {
+        return this.form.confirmP;
+      },
+
+      set(p) {
+        this.form.confirmP = p;
+        this.validateNewPassword();
+      }
+    },
+
+    canSubmit() {
+      return !!this.form.currentP && (this.isUserGenerated ? this.form.newP && this.form.newP === this.form.confirmP : true);
+    }
+
+  },
+  methods: {
+    show(show) {
+      if (show) {
+        this.reset();
+        this.$modal.show('password-modal');
+      } else {
+        this.$modal.hide('password-modal');
+      }
+    },
+    reset() {
+      this.isUserGenerated = true;
+      this.canShowMissmatchedPassword = false;
+      this.form = {
+        ...this.form,
+        currentP: null,
+        newP:     null,
+        confirmP: null
+      };
+      this.errorMessages = [];
+    },
+    passwordConfirmBlurred() {
+      this.canShowMissmatchedPassword = true;
+      this.validateNewPassword();
+    },
+    validateNewPassword() {
+      this.errorMessages = !!this.form.confirmP && (this.canShowMissmatchedPassword && this.form.newP !== this.form.confirmP) ? [this.t('prefs.account.changePassword.errors.missmatchedPassword')] : [];
+    },
+    submit(buttonCb) {
+      return this.changePassword()
+        .then(() => this.form.deleteKeys ? this.deleteKeys() : null)
+        .then(() => {
+          buttonCb(true);
+          this.show(false);
+        })
+        .catch(() => {
+          buttonCb(false);
+        });
+    },
+    changePassword() {
+      return this.$store.dispatch('rancher/request', {
```

Comment:
const res = await this.$store.dispatch('rancher/collectionAction', {
          type:       NORMAN.PRINCIPAL,
          actionName: 'search',
          opt:        { url: '/v3/principals?action=search' },
          body:       { name: str }
        });